## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 445482`
Commit: `950392067f2a7cde61541b171083ceedd3110d2a`

---
### `x86_64-linux`
<details>
  <summary>:x: 3 packages failed to build:</summary>
  <ul>
    <li>hal-hardware-analyzer</li>
    <li>python312Packages.textnets</li>
    <li>python312Packages.textnets.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 50 packages built:</summary>
  <ul>
    <li>checkov</li>
    <li>checkov.dist</li>
    <li>flattenReferencesGraph</li>
    <li>flattenReferencesGraph.dist</li>
    <li>igraph</li>
    <li>igraph.dev</li>
    <li>igraph.doc</li>
    <li>libleidenalg</li>
    <li>python312Packages.energyflow</li>
    <li>python312Packages.energyflow.dist</li>
    <li>python312Packages.explorerscript</li>
    <li>python312Packages.explorerscript.dist</li>
    <li>python312Packages.igraph</li>
    <li>python312Packages.igraph.dist</li>
    <li>python312Packages.iplotx</li>
    <li>python312Packages.iplotx.dist</li>
    <li>python312Packages.kmapper</li>
    <li>python312Packages.kmapper.dist</li>
    <li>python312Packages.leidenalg</li>
    <li>python312Packages.leidenalg.dist</li>
    <li>python312Packages.scikit-tda</li>
    <li>python312Packages.scikit-tda.dist</li>
    <li>python312Packages.skytemple-dtef</li>
    <li>python312Packages.skytemple-dtef.dist</li>
    <li>python312Packages.skytemple-files</li>
    <li>python312Packages.skytemple-files.dist</li>
    <li>python312Packages.skytemple-ssb-debugger</li>
    <li>python312Packages.skytemple-ssb-debugger.dist</li>
    <li>python313Packages.energyflow</li>
    <li>python313Packages.energyflow.dist</li>
    <li>python313Packages.explorerscript</li>
    <li>python313Packages.explorerscript.dist</li>
    <li>python313Packages.igraph</li>
    <li>python313Packages.igraph.dist</li>
    <li>python313Packages.iplotx</li>
    <li>python313Packages.iplotx.dist</li>
    <li>python313Packages.kmapper</li>
    <li>python313Packages.kmapper.dist</li>
    <li>python313Packages.leidenalg</li>
    <li>python313Packages.leidenalg.dist</li>
    <li>python313Packages.scikit-tda</li>
    <li>python313Packages.scikit-tda.dist</li>
    <li>python313Packages.skytemple-dtef</li>
    <li>python313Packages.skytemple-dtef.dist</li>
    <li>python313Packages.skytemple-files</li>
    <li>python313Packages.skytemple-files.dist</li>
    <li>python313Packages.skytemple-ssb-debugger</li>
    <li>python313Packages.skytemple-ssb-debugger.dist</li>
    <li>skytemple</li>
    <li>skytemple.dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-linux`</summary>
<details>
<summary>hal-hardware-analyzer</summary>
<pre>      |                                                                              igraph_vector_int_list_t*
/nix/store/scamh9kpqdnc2hywc804ha7w7ngx31qr-igraph-1.0.0-dev/include/igraph/igraph_paths.h:173:51: note:   initializing argument 2 of &#x27;igraph_error_t igraph_get_all_shortest_paths(const igraph_t*, const igraph_vector_t*, igraph_vector_int_list_t*, igraph_vector_int_list_t*, igraph_vector_int_t*, igraph_int_t, igraph_vs_t, igraph_neimode_t)&#x27;
  173 |     const igraph_t *graph, const igraph_vector_t *weights,
      |                            ~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~
/build/source/include/hal_core/utilities/result.h:56:5: error: could not convert &#x27;{hal::result_constructor_type::ER(), &lt;expression error&gt;}&#x27; from &#x27;&lt;brace-enclosed initializer list&gt;&#x27; to &#x27;hal::Result&lt;std::vector&lt;std::vector&lt;unsigned int&gt; &gt; &gt;&#x27;
   56 |     }
      |     ^
      |     |
      |     &lt;brace-enclosed initializer list&gt;
/build/source/plugins/graph_algorithm/src/algorithms/shortest_path.cpp:271:24: note: in expansion of macro &#x27;ERR&#x27;
  271 |                 return ERR(igraph_strerror(res));
      |                        ^~~
[1542/2353] Building CXX object plugins/boolean_influence/CMakeFiles/boolean_influence.dir/src/boolean_influence.cpp.o
[1543/2353] Building CXX object plugins/simulator/netlist_simulator_controller/CMakeFiles/netlist_simulator_controller.dir/src/dummy_engine.cpp.o
[1544/2353] Building CXX object plugins/simulator/netlist_simulator_controller/CMakeFiles/netlist_simulator_controller.dir/src/saleae_directory.cpp.o
[1545/2353] Building CXX object plugins/z3_utils/CMakeFiles/z3_utils.dir/python/python_bindings.cpp.o
[1546/2353] Building CXX object plugins/boolean_influence/CMakeFiles/boolean_influence.dir/python/python_bindings.cpp.o
[1547/2353] Building CXX object plugins/simulator/netlist_simulator_controller/CMakeFiles/netlist_simulator_controller.dir/src/plugin_netlist_simulator_controller.cpp.o
[1548/2353] Building CXX object plugins/simulator/netlist_simulator_controller/CMakeFiles/netlist_simulator_controller.dir/src/netlist_simulator_controller.cpp.o
ninja: build stopped: subcommand failed.</pre>
</details>
<details>
<summary>python312Packages.textnets</summary>
<pre>&gt;       assert net.edges[&quot;weight&quot;] == loaded.edges[&quot;weight&quot;]
E       assert [0.5440680443...43502757, ...] == [0.5440680443...43502757, ...]
E         
E         At index 1 diff: 0.5440680443502757 != 0.24303804868629442
E         Use -v to get more diff

tests/test_textnets.py:198: AssertionError
=============================== warnings summary ===============================
../../nix/store/k1jsd4b49387g8jda9ypn55vdhd9zdh1-python3.12-spacy-3.8.7/lib/python3.12/site-packages/spacy/cli/_util.py:23
  /nix/store/k1jsd4b49387g8jda9ypn55vdhd9zdh1-python3.12-spacy-3.8.7/lib/python3.12/site-packages/spacy/cli/_util.py:23: DeprecationWarning: Importing &#x27;parser.split_arg_string&#x27; is deprecated, it will only be available in &#x27;shell_completion&#x27; in Click 9.0.
    from click.parser import split_arg_string

../../nix/store/12c2mjj6jrbwqm1mypk5mm623scv3nxn-python3.12-weasel-0.4.1/lib/python3.12/site-packages/weasel/util/config.py:8
  /nix/store/12c2mjj6jrbwqm1mypk5mm623scv3nxn-python3.12-weasel-0.4.1/lib/python3.12/site-packages/weasel/util/config.py:8: DeprecationWarning: Importing &#x27;parser.split_arg_string&#x27; is deprecated, it will only be available in &#x27;shell_completion&#x27; in Click 9.0.
    from click.parser import split_arg_string

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED tests/test_textnets.py::test_textnet_save_and_load - assert [0.5440680443...43502757, ...] == [0.5440680443...43502757, ...]
================== 1 failed, 30 passed, 2 warnings in 15.77s ===================</pre>
</details>
</details>

--- a/pkgs/by-name/li/libleidenalg/package.nix
+++ b/pkgs/by-name/li/libleidenalg/package.nix
@@ -6,15 +6,15 @@
   igraph,
 }:
 
-stdenv.mkDerivation rec {
+stdenv.mkDerivation (finalAttrs: {
   pname = "libleidenalg";
-  version = "0.11.1";
+  version = "0.12.0";
 
   src = fetchFromGitHub {
     owner = "vtraag";
     repo = "libleidenalg";
-    tag = version;
-    hash = "sha256-hEES/OHvgN0yRDp5ZBZTCQfWr1j7s8NqE+Sp9WMHEEY=";
+    tag = finalAttrs.version;
+    hash = "sha256-ptfX31/1cUHLluc+Y+g28s4BEoGC0LqC9HH0cpkJRJQ=";
   };
 
   nativeBuildInputs = [
@@ -26,11 +26,11 @@ stdenv.mkDerivation rec {
   ];
 
   meta = {
-    changelog = "https://github.com/vtraag/libleidenalg/blob/${version}/CHANGELOG";
+    changelog = "https://github.com/vtraag/libleidenalg/blob/${finalAttrs.src.tag}/CHANGELOG";
     description = "C++ library of Leiden algorithm";
     homepage = "https://github.com/vtraag/libleidenalg";
     license = lib.licenses.gpl3Plus;
     maintainers = with lib.maintainers; [ dotlambda ];
     platforms = lib.platforms.all;
   };
-}
+})

--- a/pkgs/development/python-modules/igraph/default.nix
+++ b/pkgs/development/python-modules/igraph/default.nix
@@ -15,7 +15,7 @@
 
 buildPythonPackage rec {
   pname = "igraph";
-  version = "0.11.9";
+  version = "1.0.0";
 
   pyproject = true;
 
@@ -27,7 +27,7 @@ buildPythonPackage rec {
       # export-subst prevents reproducability
       rm $out/.git_archival.json
     '';
-    hash = "sha256-rmIICiIyEr5JCmkDAzcdisVaaKDraTQEquPHjK4d7oU=";
+    hash = "sha256-Y7ZQ1yNoD8A5b6c92OGz9Unietdg1uNt/Za6nxdCSP0=";
   };
 
   postPatch = ''

--- a/pkgs/development/python-modules/leidenalg/default.nix
+++ b/pkgs/development/python-modules/leidenalg/default.nix
@@ -6,26 +6,23 @@
   igraph,
   igraph-c,
   libleidenalg,
-  pythonOlder,
   setuptools-scm,
   unittestCheckHook,
 }:
 
 buildPythonPackage rec {
   pname = "leidenalg";
-  version = "0.10.2";
+  version = "0.11.0";
   pyproject = true;
-
-  disabled = pythonOlder "3.7";
 
   src = fetchFromGitHub {
     owner = "vtraag";
     repo = "leidenalg";
     tag = version;
-    hash = "sha256-oaTV+BIB/YQBWKrVXuiIEMH/1MxPxeHhjUzbmxt6hlw=";
+    hash = "sha256-u4xW1gsWDpbsFVLlOIiPZtpw9t4iFBC7fzwn04flev8=";
   };
 
-  nativeBuildInputs = [ setuptools-scm ];
+  build-system = [ setuptools-scm ];
 
   buildInputs = [
     igraph-c
@@ -34,18 +31,18 @@ buildPythonPackage rec {
 
   propagatedBuildInputs = [ igraph ];
 
-  checkInputs = [
+  nativeCheckInputs = [
     ddt
     unittestCheckHook
   ];
 
   pythonImportsCheck = [ "leidenalg" ];
 
-  meta = with lib; {
-    changelog = "https://github.com/vtraag/leidenalg/blob/${version}/CHANGELOG";
+  meta = {
+    changelog = "https://github.com/vtraag/leidenalg/blob/${src.tag}/CHANGELOG";
     description = "Implementation of the Leiden algorithm for various quality functions to be used with igraph in Python";
     homepage = "https://github.com/vtraag/leidenalg";
-    license = licenses.gpl3Only;
-    maintainers = with maintainers; [ jboy ];
+    license = lib.licenses.gpl3Plus;
+    maintainers = with lib.maintainers; [ jboy ];
   };
 }

